### PR TITLE
Clarify setting `CompressionOptions::max_dict_bytes > 0` will charge block cache

### PR DESCRIPTION
**Pull request by ajkr**
Add it to the API comment.

### CODE REVIEW

File: include/rocksdb/advanced_options.h [hunk 1]
```diff
@@ -105,9 +105,13 @@ struct CompressionOptions {
   //
   // When compression dictionary is disabled, we compress and write each block
   // before buffering data for the next one. When compression dictionary is
-  // enabled, we buffer all SST file data in-memory so we can sample it, as data
+  // enabled, we buffer SST file data in-memory so we can sample it, as data
   // can only be compressed and written after the dictionary has been finalized.
-  // So users of this feature may see increased memory usage.
+  //
+  // The amount of data buffered can be limited by `max_dict_buffer_bytes`. This
+  // buffered memory is charged to block cache when available. If block cache
```
Comment (hx235) [inline include/rocksdb/advanced_options.h:112]: Do you see a need of specifying "when available"? I am thinking something like "..when the block cache is available".

File: include/rocksdb/advanced_options.h [hunk 2]
```diff
@@ -105,9 +105,13 @@ struct CompressionOptions {
...
+  // insertion fails with `Status::Incomplete` (i.e., it is full), we finalize
```
Comment (hx235) [inline include/rocksdb/advanced_options.h:113]: A terminology usage about "cache full": do we always need to say `strict_capacity_limit=true` along with saying "cache is full" or does "cache full" already indicate `strict_capacity_limit=true`?

**Comment by hx235** [comment]
Left some minor suggestion - otherwise LGTM!

File: include/rocksdb/advanced_options.h (hunk 1 above)
Comment (ajkr) [reply include/rocksdb/advanced_options.h]: Yes, it felt slightly ambiguous to me too. Done, with adjusted phrasing.

**Review by hx235 — APPROVED** [review]

File: include/rocksdb/advanced_options.h (hunk 2 above)
Comment (ajkr) [reply include/rocksdb/advanced_options.h]: I wondered about this too. Interestingly `Cache::Insert` does mention `strict_capacity_limit`: https://github.com/facebook/rocksdb/blob/f72c834eab08f6e72780d8fead6039acfe565f44/include/rocksdb/cache.h#L243-L244.

Despite that I still felt `strict_capacity_limit` is just the name of an option for particular implementations of `Cache`...

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

File: include/rocksdb/advanced_options.h (hunk 2 above)
Comment (ajkr) [reply include/rocksdb/advanced_options.h]: > or does "cache full" already indicate strict_capacity_limit=true?

Probably not very clearly. But with the context of returning `Status::Incomplete` it does.

Comment (hx235) [reply include/rocksdb/advanced_options.h]: I mentioned `strict_capacity_limit ` in HISTORY.md and filter construction related API anyway. Maybe we should all go for a general expression like "when the cache under a strict capacity limit" but then I realized the difference is jut removing the underscore ¯\_(ツ)_/¯

**Pull request closed without merging**